internal/resources/iampolicy/scope.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	clustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	clustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/clustergroup"
+	namespacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/namespace"
+	organizationmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/organization"
+	workspacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/workspace"
+)
+
+type scopeType int
+
+const (
+	unknown scopeType = iota
+	organization
+	clusterGroup
+	cluster
+	workspace
+	namespace
+)
+
+var (
+	stringToScope = map[string]scopeType{
+		orgKey:          organization,
+		clusterGroupKey: clusterGroup,
+		clusterKey:      cluster,
+		workspaceKey:    workspace,
+		namespaceKey:    namespace,
+	}
+)
+
+func (s scopeType) String() string {
+	return [...]string{
+		"unknown",
+		orgKey,
+		clusterGroupKey,
+		clusterKey,
+		workspaceKey,
+		namespaceKey,
+	}[s]
+}
+
+func constructScopeFullName(s scopeType, d *schema.ResourceData) interface{} {
+	scopeData, ok := d.GetOk(scopeKey)
+	if !ok {
+		return nil
+	}
+
+	scopeDataList, _ := scopeData.([]interface{})
+	if len(scopeDataList) < 1 {
+		return nil
+	}
+
+	scopeDataMap, _ := scopeDataList[0].(map[string]interface{})
+
+	val, ok := scopeDataMap[s.String()]
+	if !ok {
+		return nil
+	}
+
+	valList := val.([]interface{})
+	if len(valList) < 1 {
+		return nil
+	}
+
+	valMap, _ := valList[0].(map[string]interface{})
+
+	switch s {
+	case organization:
+		return &organizationmodel.VmwareTanzuManageV1alpha1OrganizationFullName{
+			OrgID: valMap[orgIDKey].(string),
```

Comment:
do you setPrimitive helper function here and in other places

Reply:
Done